server/src/main/java/org/elasticsearch/script/TemplateScript.java

```diff
@@ -35,4 +37,8 @@ public TemplateScript(Map<String, Object> params) {
     }
 
     public static final ScriptContext<Factory> CONTEXT = new ScriptContext<>("template", Factory.class);
+
+    // Remove compilation rate limit for ingest
```

Comment:
I think this deserves a larger comment. This will be confusing enough just based on the naming here, and the fact that the same script class is used by 2 different contexts. It would be good to describe the reasons we haven't created another script class for this (complexity of implementation in mustache script engine, possible but not worth effort for now). The reason for having a separate context is also worth still mentioning, as you have, I would just add this additional info.

Reply:
Makes sense. Done.